src/components/menu/API.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import {Trans, withTranslation, WithTranslation} from "react-i18next";
+import {IAlert} from "../../types";
+
+
+interface IState {
+    alerts: IAlert[];
+}
+
+class API extends React.Component<WithTranslation, IState> {
+
+    state = {alerts: []};
+
+    public render() {
+    
+        return (
+            <div>
+                <div className="content box" id="API">
+                    <h3><Trans>API</Trans></h3>
+                        <h4>Preset</h4>
+                        <h5>Preset</h5>
+                            <p>Request the selected preset.</p>
+                            <dl>
+                                <dt>Endpoint</dt>
+                                <dd>/api/preset</dd>   
+                                <dt>Request Parameters</dt>
+                                <dd>id</dd>
+                                <dt>Example Request</dt>
+                                <dd>https://aoe2cm.net/api/preset/simple</dd>
+                                <dd>
+                                <ul>
+                                    <li>TECHNICAL = 0</li>
+                                    <li>AOK = 1</li>
+                                    <li>CONQUERORS = 2</li>
+                                    <li>FORGOTTEN = 3</li>
+                                    <li>AFRICAN_KINGDOMS = 4</li>
+                                    <li>RISE_OF_RAJAS = 5</li>
+                                    <li>DEFINITIVE_EDITION = 6</li>
+                                    <li>LORDS_OF_THE_WEST = 7</li>
+                                    </ul>
+                                </dd>
+                            </dl>
+
+                            <h5>Add Preset</h5>
+                            <p>Request the selected preset.</p>
+                            <dl>
+                                <dt>Endpoint</dt>
+                                <dd>/api/preset</dd>   
+                                <dt>Request Parameters</dt>
+                                <dd>No Clue</dd>
```

Comment:
Basically, you send a JSON string in the following format: 
```json
{
  "preset": {
    "name": string,
    "encodedCivilisations": string,
    "turns": Turn[]
  }
}
```
and get back `{"status": "ok", "presetId": "abcdef"}`

Reply:
should i copy paste this into the <dd>No clue</dd>.

Reply:
Depends on how you would like to do it :shrug:

Reply:
I would like to put everything there so it looks like an url if possible
https://aoe2.net/api/leaderboard?game=XXXX&leaderboard_id=X&start=X&count=X

Reply:
Well, it's a POST request, which means you don't have query parameters in the url :shrug: